test(core): Dynamically allocate ports for Python integration tests

ivov opened the pull request:
## Summary

Locally whenever you have an n8n instance running, the Python integration test ports are taken so tests hang indefinitely with no indication of what the issue is. This PR dynamically allocates ports as I keep running into this.

## Related Linear tickets, Github issues, and Community forum posts

n/a


## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
   **Remember, the title automatically goes into the changelog.
   Use `(no-changelog)` otherwise.**
-->
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. <!--
   A bug is not considered fixed, unless a test is added to prevent it from happening again.
   A feature is not complete without tests.
-->
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

ivov committed:
test(core): Dynamically allocate ports for Python integration tests

Cadiac commented on packages/@n8n/task-runner-python/tests/integration/test_port.py:
```diff
@@ -0,0 +1,11 @@
+def find_free_port() -> int:
+    """Find and return a free port on localhost."""
+    
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
```
is there a risk that a parallel test or something could steal the port before we take it into use at the test, since we close the socket after this `with`?

Probably not a real issue in practice though

ivov replied:
Thanks for bringing this up! We don't want to introduce flakiness, no matter how unlikely. Please see [`6812a6a`](https://github.com/n8n-io/n8n/pull/20825/commits/6812a6a22b841540c51f3ad15890457e37895370)

ivov committed:
Fix port allocation race condition in tests

Use OS-assigned ports (bind to port 0) to prevent race condition when
running tests in parallel. The old approach closed the socket after
finding a free port, allowing another process to steal it before use.

Cadiac commented on packages/@n8n/task-runner-python/src/health_check_server.py:
```diff
@@ -20,8 +20,9 @@ async def start(self, config: HealthCheckConfig) -> None:
+            actual_port = self.server.sockets[0].getsockname()[1] # for OS-assigned port in tests 
             self.logger.info(
-                f"Health check server listening on {config.host}, port {config.port}"
+                f"Health check server listening on {config.host}, port {actual_port}"
```
Did we have a way to control the loglevel of this, is there a risk that this logging happens on loglevel above `info` and we never log this -> it never ends up on `stdout_buffer` -> tests fail at `_wait_for_health_check_port` ? 

I think there's no such risk here, we only monitor the port logline at tests and they're run with level where this gets printed?

ivov committed:
Harcode info level logging in tests

ivov committed:
Use constant

Joffcom commented:
Got released with [`n8n@1.117.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.117.0)
